tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -22,6 +24,162 @@ readonly KBS_NS="coco-tenant"
+		--policy-file "$file"
+}
+
+# Set resource data.
+#
+# Parameters:
+#	$1 - repository name (optional)
+#	$2 - resource type (mandatory)
+#	$3 - tag (mandatory)
+#	$4 - resource data
+#
+# Note: kbs-client will be installed if not yet.
+#
+kbs_set_resource() {
+	local repository="${1:-}"
+	local type="${2:-}"
+	local tag="${3:-}"
+	local data="${4:-}"
+	local file
+	local rc=0
+
+	if [ -z "$data" ]; then
+		echo "ERROR: missing data parameter"
+		return 1
+	fi
+
+	file=$(mktemp -t kbs-resource-XXXXX)
+	echo "$data" > "$file"
+
+	kbs_set_resource_from_file "$repository" "$type" "$tag" "$file" || \
+		rc=$?
+
+	rm -f "$file"
+	return $rc
+}
+
+# Set resource, read data from file.
+#
+# Parameters:
+#	$1 - repository name (optional)
+#	$2 - resource type (mandatory)
+#	$3 - tag (mandatory)
+#	$4 - resource data
+#
+# Note: kbs-client will be installed if not yet.
+#
+kbs_set_resource_from_file() {
+	local repository="${1:-}"
+	local type="${2:-}"
+	local tag="${3:-}"
+	local file="${4:-}"
+
+	if [[ -z "$type" || -z "$tag" ]]; then
+		echo "ERROR: missing type='$type' and/or tag='$tag' parameters"
+		return 1
+	elif [ ! -f "$file" ]; then
+		echo "ERROR: resource file '$file' does not exist"
+		return 1
+	fi
+
+	local path=""
+	[ -n "$repository" ] && path+="${repository}/"
+	path+="${type}/"
+	path+="${tag}"
+
+	kbs-client --url "$(kbs_k8s_svc_http_addr)" config \
+		--auth-private-key "$(kbs_get_private_key)" set-resource \
+		--path "$path" --resource-file "$file"
+}
+
+# Return the private key file used for CLI authentication.
+#
+kbs_get_private_key() {
+	echo "$COCO_KBS_DIR/kbs/config/kubernetes/base/kbs.key"
+}
+
+# Build and install the kbs-client binary, unless it is already present.
+#
+#
+kbs_install_cli() {
+	command -v kbs-client >/dev/null && return
+
+	if ! command -v apt >/dev/null; then
+		echo "ERROR: running on unsupported distro"
+		return 1
+	fi
+
+	local pkgs="build-essential"
+
+	sudo apt-get update -y
+	# shellcheck disable=2086
+	sudo apt-get install -y $pkgs
+
+	# Mininum required version to build the client (read from versions.yaml)
+	local rust_version
+	ensure_yq
+	rust_version=$(get_from_kata_deps "externals.coco-kbs.rust")
+
+	if ! command -v rustc >/dev/null; then
+		# Currently kata version from version.yaml is 1.72.0
+		# which doesn't match the requirement, so let's pass
+		# the required version.
+		"${kubernetes_dir}/../../install_rust.sh" "${rust_version}"
+
+		# shellcheck disable=1091
+		source "$HOME/.cargo/env"
+	else
+		# We don't want to mess with installation on bare-metal so
+		# if rust is installed then just check it's >= the required
+		# version.
+		#
+		local current_rust_version
+		current_rust_version="$(rustc --version | cut -d' ' -f2)"
+		if ! version_greater_than_equal "${current_rust_version}" \
+			"${rust_version}"; then
+			echo "ERROR: installed rust $current_rust_version < $rust_version (required)"
+			return 1
+		fi
+	fi
```

Comment:
Suggestion: This block for validating the rust version could probably get moved to a separate function to keep the functionality of kbs_install_ci easy to read.

Reply:
I didn't move to `common.bash` because I realized some troubles that I wouldn't have time to work out now. Instead moved to an internal function within `confidential_kbs.sh`.